// @portone/server-sdk@0.8.1/dist/generated/index.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AccountClient", {
  enumerable: true,
  get: function () {
    return _client.AccountClient;
  }
});
Object.defineProperty(exports, "AccountTransferClient", {
  enumerable: true,
  get: function () {
    return _client2.AccountTransferClient;
  }
});
Object.defineProperty(exports, "AuthClient", {
  enumerable: true,
  get: function () {
    return _client3.AuthClient;
  }
});
Object.defineProperty(exports, "BillingKeyClient", {
  enumerable: true,
  get: function () {
    return _client4.BillingKeyClient;
  }
});
Object.defineProperty(exports, "BulkPayoutClient", {
  enumerable: true,
  get: function () {
    return _client5.BulkPayoutClient;
  }
});
Object.defineProperty(exports, "CashReceiptClient", {
  enumerable: true,
  get: function () {
    return _client6.CashReceiptClient;
  }
});
Object.defineProperty(exports, "IdentityVerificationClient", {
  enumerable: true,
  get: function () {
    return _client7.IdentityVerificationClient;
  }
});
Object.defineProperty(exports, "PartnerClient", {
  enumerable: true,
  get: function () {
    return _client8.PartnerClient;
  }
});
Object.defineProperty(exports, "PartnerSettlementClient", {
  enumerable: true,
  get: function () {
    return _client9.PartnerSettlementClient;
  }
});
Object.defineProperty(exports, "PaymentClient", {
  enumerable: true,
  get: function () {
    return _client10.PaymentClient;
  }
});
Object.defineProperty(exports, "PaymentScheduleClient", {
  enumerable: true,
  get: function () {
    return _client11.PaymentScheduleClient;
  }
});
Object.defineProperty(exports, "PayoutClient", {
  enumerable: true,
  get: function () {
    return _client12.PayoutClient;
  }
});
Object.defineProperty(exports, "PgSpecificClient", {
  enumerable: true,
  get: function () {
    return _client13.PgSpecificClient;
  }
});
Object.defineProperty(exports, "PlatformClient", {
  enumerable: true,
  get: function () {
    return _client14.PlatformClient;
  }
});
Object.defineProperty(exports, "PolicyClient", {
  enumerable: true,
  get: function () {
    return _client15.PolicyClient;
  }
});
Object.defineProperty(exports, "PortOneClient", {
  enumerable: true,
  get: function () {
    return _client16.PortOneClient;
  }
});
Object.defineProperty(exports, "PromotionClient", {
  enumerable: true,
  get: function () {
    return _client17.PromotionClient;
  }
});
Object.defineProperty(exports, "RestError", {
  enumerable: true,
  get: function () {
    return _RestError.RestError;
  }
});
Object.defineProperty(exports, "TransferClient", {
  enumerable: true,
  get: function () {
    return _client18.TransferClient;
  }
});
exports.platform = exports.pgSpecific = exports.payment = exports.identityVerification = exports.common = exports.auth = void 0;
var _RestError = require("./RestError.cjs");
var _auth = _interopRequireWildcard(require("./auth/index.cjs"));
exports.auth = _auth;
var _platform = _interopRequireWildcard(require("./platform/index.cjs"));
exports.platform = _platform;
var _identityVerification = _interopRequireWildcard(require("./identityVerification/index.cjs"));
exports.identityVerification = _identityVerification;
var _payment = _interopRequireWildcard(require("./payment/index.cjs"));
exports.payment = _payment;
var _pgSpecific = _interopRequireWildcard(require("./pgSpecific/index.cjs"));
exports.pgSpecific = _pgSpecific;
var _common = _interopRequireWildcard(require("./common/index.cjs"));
exports.common = _common;
var _client = require("./platform/account/client.cjs");
var _client2 = require("./platform/accountTransfer/client.cjs");
var _client3 = require("./auth/client.cjs");
var _client4 = require("./payment/billingKey/client.cjs");
var _client5 = require("./platform/bulkPayout/client.cjs");
var _client6 = require("./payment/cashReceipt/client.cjs");
var _client7 = require("./identityVerification/client.cjs");
var _client8 = require("./platform/partner/client.cjs");
var _client9 = require("./platform/partnerSettlement/client.cjs");
var _client10 = require("./payment/client.cjs");
var _client11 = require("./payment/paymentSchedule/client.cjs");
var _client12 = require("./platform/payout/client.cjs");
var _client13 = require("./pgSpecific/client.cjs");
var _client14 = require("./platform/client.cjs");
var _client15 = require("./platform/policy/client.cjs");
var _client16 = require("./client.cjs");
var _client17 = require("./payment/promotion/client.cjs");
var _client18 = require("./platform/transfer/client.cjs");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
