// @portone/server-sdk@0.8.1/dist/generated/platform/accountTransfer/client.mjs
import { AccountTransferError } from "./AccountTransferError.mjs";
import { USER_AGENT } from "../../../client.mjs";
export function AccountTransferClient(init) {
  const baseUrl = init.baseUrl ?? "https://api.portone.io";
  const secret = init.secret;
  return {
    getPlatformAccountTransfers: async (options) => {
      const isForTest = options?.isForTest;
      const page = options?.page;
      const filter = options?.filter;
      const requestBody = JSON.stringify({
        isForTest,
        page,
        filter
      });
      const query = [
        ["requestBody", requestBody]
      ].flatMap(([key, value]) => value == null ? [] : `${key}=${encodeURIComponent(value)}`).join("&");
      const response = await fetch(
        new URL(`/platform/account-transfers?${query}`, baseUrl),
        {
          method: "GET",
          headers: {
            Authorization: `PortOne ${secret}`,
            "User-Agent": USER_AGENT
          }
        }
      );
      if (!response.ok) {
        throw new GetPlatformAccountTransfersError(await response.json());
      }
      return response.json();
    }
  };
}
export class GetPlatformAccountTransfersError extends AccountTransferError {
  /** @ignore */
  constructor(data) {
    super(data);
    Object.setPrototypeOf(this, GetPlatformAccountTransfersError.prototype);
    this.name = "GetPlatformAccountTransfersError";
  }
}
