// @portone/server-sdk@0.8.1/dist/generated/platform/index.cjs
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  PlatformError: true,
  policy: true,
  partner: true,
  transfer: true,
  partnerSettlement: true,
  payout: true,
  bulkPayout: true,
  account: true,
  accountTransfer: true
};
Object.defineProperty(exports, "PlatformError", {
  enumerable: true,
  get: function () {
    return _PlatformError.PlatformError;
  }
});
exports.transfer = exports.policy = exports.payout = exports.partnerSettlement = exports.partner = exports.bulkPayout = exports.accountTransfer = exports.account = void 0;
var _PlatformError = require("./PlatformError.cjs");
var _CancelPlatformAdditionalFeePolicyScheduleResponse = require("./CancelPlatformAdditionalFeePolicyScheduleResponse.cjs");
Object.keys(_CancelPlatformAdditionalFeePolicyScheduleResponse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _CancelPlatformAdditionalFeePolicyScheduleResponse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _CancelPlatformAdditionalFeePolicyScheduleResponse[key];
    }
  });
});
var _CancelPlatformContractScheduleResponse = require("./CancelPlatformContractScheduleResponse.cjs");
Object.keys(_CancelPlatformContractScheduleResponse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _CancelPlatformContractScheduleResponse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _CancelPlatformContractScheduleResponse[key];
    }
  });
});
var _CancelPlatformDiscountSharePolicyScheduleResponse = require("./CancelPlatformDiscountSharePolicyScheduleResponse.cjs");
Object.keys(_CancelPlatformDiscountSharePolicyScheduleResponse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _CancelPlatformDiscountSharePolicyScheduleResponse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _CancelPlatformDiscountSharePolicyScheduleResponse[key];
    }
  });
});
var _CancelPlatformPartnerScheduleResponse = require("./CancelPlatformPartnerScheduleResponse.cjs");
Object.keys(_CancelPlatformPartnerScheduleResponse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _CancelPlatformPartnerScheduleResponse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _CancelPlatformPartnerScheduleResponse[key];
    }
  });
});
var _DateRange = require("./DateRange.cjs");
Object.keys(_DateRange).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _DateRange[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _DateRange[key];
    }
  });
});
var _DayOfWeek = require("./DayOfWeek.cjs");
Object.keys(_DayOfWeek).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _DayOfWeek[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _DayOfWeek[key];
    }
  });
});
var _MonthDay = require("./MonthDay.cjs");
Object.keys(_MonthDay).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _MonthDay[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _MonthDay[key];
    }
  });
});
var _Platform = require("./Platform.cjs");
Object.keys(_Platform).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _Platform[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _Platform[key];
    }
  });
});
var _PlatformAccount = require("./PlatformAccount.cjs");
Object.keys(_PlatformAccount).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformAccount[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformAccount[key];
    }
  });
});
var _PlatformAccountStatus = require("./PlatformAccountStatus.cjs");
Object.keys(_PlatformAccountStatus).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformAccountStatus[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformAccountStatus[key];
    }
  });
});
var _PlatformAccountVerificationAlreadyUsedError = require("./PlatformAccountVerificationAlreadyUsedError.cjs");
Object.keys(_PlatformAccountVerificationAlreadyUsedError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformAccountVerificationAlreadyUsedError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformAccountVerificationAlreadyUsedError[key];
    }
  });
});
var _PlatformAccountVerificationFailedError = require("./PlatformAccountVerificationFailedError.cjs");
Object.keys(_PlatformAccountVerificationFailedError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformAccountVerificationFailedError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformAccountVerificationFailedError[key];
    }
  });
});
var _PlatformAccountVerificationNotFoundError = require("./PlatformAccountVerificationNotFoundError.cjs");
Object.keys(_PlatformAccountVerificationNotFoundError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformAccountVerificationNotFoundError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformAccountVerificationNotFoundError[key];
    }
  });
});
var _PlatformAdditionalFeePolicy = require("./PlatformAdditionalFeePolicy.cjs");
Object.keys(_PlatformAdditionalFeePolicy).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformAdditionalFeePolicy[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformAdditionalFeePolicy[key];
    }
  });
});
var _PlatformAdditionalFeePolicyNotFoundError = require("./PlatformAdditionalFeePolicyNotFoundError.cjs");
Object.keys(_PlatformAdditionalFeePolicyNotFoundError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformAdditionalFeePolicyNotFoundError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformAdditionalFeePolicyNotFoundError[key];
    }
  });
});
var _PlatformAdditionalFeePolicyScheduleAlreadyExistsError = require("./PlatformAdditionalFeePolicyScheduleAlreadyExistsError.cjs");
Object.keys(_PlatformAdditionalFeePolicyScheduleAlreadyExistsError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformAdditionalFeePolicyScheduleAlreadyExistsError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformAdditionalFeePolicyScheduleAlreadyExistsError[key];
    }
  });
});
var _PlatformArchivedAdditionalFeePolicyError = require("./PlatformArchivedAdditionalFeePolicyError.cjs");
Object.keys(_PlatformArchivedAdditionalFeePolicyError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformArchivedAdditionalFeePolicyError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformArchivedAdditionalFeePolicyError[key];
    }
  });
});
var _PlatformArchivedContractError = require("./PlatformArchivedContractError.cjs");
Object.keys(_PlatformArchivedContractError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformArchivedContractError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformArchivedContractError[key];
    }
  });
});
var _PlatformArchivedDiscountSharePolicyError = require("./PlatformArchivedDiscountSharePolicyError.cjs");
Object.keys(_PlatformArchivedDiscountSharePolicyError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformArchivedDiscountSharePolicyError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformArchivedDiscountSharePolicyError[key];
    }
  });
});
var _PlatformArchivedPartnerError = require("./PlatformArchivedPartnerError.cjs");
Object.keys(_PlatformArchivedPartnerError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformArchivedPartnerError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformArchivedPartnerError[key];
    }
  });
});
var _PlatformArchivedPartnersCannotBeScheduledError = require("./PlatformArchivedPartnersCannotBeScheduledError.cjs");
Object.keys(_PlatformArchivedPartnersCannotBeScheduledError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformArchivedPartnersCannotBeScheduledError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformArchivedPartnersCannotBeScheduledError[key];
    }
  });
});
var _PlatformContact = require("./PlatformContact.cjs");
Object.keys(_PlatformContact).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformContact[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformContact[key];
    }
  });
});
var _PlatformContract = require("./PlatformContract.cjs");
Object.keys(_PlatformContract).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformContract[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformContract[key];
    }
  });
});
var _PlatformContractNotFoundError = require("./PlatformContractNotFoundError.cjs");
Object.keys(_PlatformContractNotFoundError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformContractNotFoundError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformContractNotFoundError[key];
    }
  });
});
var _PlatformContractScheduleAlreadyExistsError = require("./PlatformContractScheduleAlreadyExistsError.cjs");
Object.keys(_PlatformContractScheduleAlreadyExistsError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformContractScheduleAlreadyExistsError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformContractScheduleAlreadyExistsError[key];
    }
  });
});
var _PlatformCurrencyNotSupportedError = require("./PlatformCurrencyNotSupportedError.cjs");
Object.keys(_PlatformCurrencyNotSupportedError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformCurrencyNotSupportedError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformCurrencyNotSupportedError[key];
    }
  });
});
var _PlatformDiscountSharePolicy = require("./PlatformDiscountSharePolicy.cjs");
Object.keys(_PlatformDiscountSharePolicy).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformDiscountSharePolicy[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformDiscountSharePolicy[key];
    }
  });
});
var _PlatformDiscountSharePolicyFilterOptions = require("./PlatformDiscountSharePolicyFilterOptions.cjs");
Object.keys(_PlatformDiscountSharePolicyFilterOptions).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformDiscountSharePolicyFilterOptions[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformDiscountSharePolicyFilterOptions[key];
    }
  });
});
var _PlatformDiscountSharePolicyNotFoundError = require("./PlatformDiscountSharePolicyNotFoundError.cjs");
Object.keys(_PlatformDiscountSharePolicyNotFoundError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformDiscountSharePolicyNotFoundError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformDiscountSharePolicyNotFoundError[key];
    }
  });
});
var _PlatformDiscountSharePolicyScheduleAlreadyExistsError = require("./PlatformDiscountSharePolicyScheduleAlreadyExistsError.cjs");
Object.keys(_PlatformDiscountSharePolicyScheduleAlreadyExistsError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformDiscountSharePolicyScheduleAlreadyExistsError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformDiscountSharePolicyScheduleAlreadyExistsError[key];
    }
  });
});
var _PlatformFee = require("./PlatformFee.cjs");
Object.keys(_PlatformFee).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformFee[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformFee[key];
    }
  });
});
var _PlatformFeeInput = require("./PlatformFeeInput.cjs");
Object.keys(_PlatformFeeInput).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformFeeInput[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformFeeInput[key];
    }
  });
});
var _PlatformFixedAmountFee = require("./PlatformFixedAmountFee.cjs");
Object.keys(_PlatformFixedAmountFee).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformFixedAmountFee[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformFixedAmountFee[key];
    }
  });
});
var _PlatformFixedRateFee = require("./PlatformFixedRateFee.cjs");
Object.keys(_PlatformFixedRateFee).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformFixedRateFee[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformFixedRateFee[key];
    }
  });
});
var _PlatformInsufficientDataToChangePartnerTypeError = require("./PlatformInsufficientDataToChangePartnerTypeError.cjs");
Object.keys(_PlatformInsufficientDataToChangePartnerTypeError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformInsufficientDataToChangePartnerTypeError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformInsufficientDataToChangePartnerTypeError[key];
    }
  });
});
var _PlatformInvalidSettlementFormulaError = require("./PlatformInvalidSettlementFormulaError.cjs");
Object.keys(_PlatformInvalidSettlementFormulaError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformInvalidSettlementFormulaError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformInvalidSettlementFormulaError[key];
    }
  });
});
var _PlatformNotEnabledError = require("./PlatformNotEnabledError.cjs");
Object.keys(_PlatformNotEnabledError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformNotEnabledError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformNotEnabledError[key];
    }
  });
});
var _PlatformOrderSettlementAmount = require("./PlatformOrderSettlementAmount.cjs");
Object.keys(_PlatformOrderSettlementAmount).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformOrderSettlementAmount[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformOrderSettlementAmount[key];
    }
  });
});
var _PlatformPartner = require("./PlatformPartner.cjs");
Object.keys(_PlatformPartner).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartner[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartner[key];
    }
  });
});
var _PlatformPartnerBusinessStatus = require("./PlatformPartnerBusinessStatus.cjs");
Object.keys(_PlatformPartnerBusinessStatus).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartnerBusinessStatus[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartnerBusinessStatus[key];
    }
  });
});
var _PlatformPartnerContractSummary = require("./PlatformPartnerContractSummary.cjs");
Object.keys(_PlatformPartnerContractSummary).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartnerContractSummary[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartnerContractSummary[key];
    }
  });
});
var _PlatformPartnerFilterInput = require("./PlatformPartnerFilterInput.cjs");
Object.keys(_PlatformPartnerFilterInput).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartnerFilterInput[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartnerFilterInput[key];
    }
  });
});
var _PlatformPartnerFilterInputKeyword = require("./PlatformPartnerFilterInputKeyword.cjs");
Object.keys(_PlatformPartnerFilterInputKeyword).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartnerFilterInputKeyword[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartnerFilterInputKeyword[key];
    }
  });
});
var _PlatformPartnerFilterOptions = require("./PlatformPartnerFilterOptions.cjs");
Object.keys(_PlatformPartnerFilterOptions).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartnerFilterOptions[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartnerFilterOptions[key];
    }
  });
});
var _PlatformPartnerNotFoundError = require("./PlatformPartnerNotFoundError.cjs");
Object.keys(_PlatformPartnerNotFoundError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartnerNotFoundError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartnerNotFoundError[key];
    }
  });
});
var _PlatformPartnerScheduleAlreadyExistsError = require("./PlatformPartnerScheduleAlreadyExistsError.cjs");
Object.keys(_PlatformPartnerScheduleAlreadyExistsError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartnerScheduleAlreadyExistsError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartnerScheduleAlreadyExistsError[key];
    }
  });
});
var _PlatformPartnerSchedulesAlreadyExistError = require("./PlatformPartnerSchedulesAlreadyExistError.cjs");
Object.keys(_PlatformPartnerSchedulesAlreadyExistError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartnerSchedulesAlreadyExistError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartnerSchedulesAlreadyExistError[key];
    }
  });
});
var _PlatformPartnerStatus = require("./PlatformPartnerStatus.cjs");
Object.keys(_PlatformPartnerStatus).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartnerStatus[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartnerStatus[key];
    }
  });
});
var _PlatformPartnerTaxationType = require("./PlatformPartnerTaxationType.cjs");
Object.keys(_PlatformPartnerTaxationType).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartnerTaxationType[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartnerTaxationType[key];
    }
  });
});
var _PlatformPartnerType = require("./PlatformPartnerType.cjs");
Object.keys(_PlatformPartnerType).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartnerType[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartnerType[key];
    }
  });
});
var _PlatformPartnerTypeBusiness = require("./PlatformPartnerTypeBusiness.cjs");
Object.keys(_PlatformPartnerTypeBusiness).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartnerTypeBusiness[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartnerTypeBusiness[key];
    }
  });
});
var _PlatformPartnerTypeNonWhtPayer = require("./PlatformPartnerTypeNonWhtPayer.cjs");
Object.keys(_PlatformPartnerTypeNonWhtPayer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartnerTypeNonWhtPayer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartnerTypeNonWhtPayer[key];
    }
  });
});
var _PlatformPartnerTypeWhtPayer = require("./PlatformPartnerTypeWhtPayer.cjs");
Object.keys(_PlatformPartnerTypeWhtPayer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPartnerTypeWhtPayer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPartnerTypeWhtPayer[key];
    }
  });
});
var _PlatformPayer = require("./PlatformPayer.cjs");
Object.keys(_PlatformPayer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPayer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPayer[key];
    }
  });
});
var _PlatformPayoutMethod = require("./PlatformPayoutMethod.cjs");
Object.keys(_PlatformPayoutMethod).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPayoutMethod[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPayoutMethod[key];
    }
  });
});
var _PlatformPayoutStatusStats = require("./PlatformPayoutStatusStats.cjs");
Object.keys(_PlatformPayoutStatusStats).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformPayoutStatusStats[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformPayoutStatusStats[key];
    }
  });
});
var _PlatformProperties = require("./PlatformProperties.cjs");
Object.keys(_PlatformProperties).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformProperties[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformProperties[key];
    }
  });
});
var _PlatformRoundType = require("./PlatformRoundType.cjs");
Object.keys(_PlatformRoundType).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformRoundType[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformRoundType[key];
    }
  });
});
var _PlatformSettlementCycle = require("./PlatformSettlementCycle.cjs");
Object.keys(_PlatformSettlementCycle).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementCycle[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementCycle[key];
    }
  });
});
var _PlatformSettlementCycleDatePolicy = require("./PlatformSettlementCycleDatePolicy.cjs");
Object.keys(_PlatformSettlementCycleDatePolicy).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementCycleDatePolicy[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementCycleDatePolicy[key];
    }
  });
});
var _PlatformSettlementCycleInput = require("./PlatformSettlementCycleInput.cjs");
Object.keys(_PlatformSettlementCycleInput).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementCycleInput[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementCycleInput[key];
    }
  });
});
var _PlatformSettlementCycleMethod = require("./PlatformSettlementCycleMethod.cjs");
Object.keys(_PlatformSettlementCycleMethod).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementCycleMethod[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementCycleMethod[key];
    }
  });
});
var _PlatformSettlementCycleMethodDaily = require("./PlatformSettlementCycleMethodDaily.cjs");
Object.keys(_PlatformSettlementCycleMethodDaily).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementCycleMethodDaily[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementCycleMethodDaily[key];
    }
  });
});
var _PlatformSettlementCycleMethodDailyInput = require("./PlatformSettlementCycleMethodDailyInput.cjs");
Object.keys(_PlatformSettlementCycleMethodDailyInput).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementCycleMethodDailyInput[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementCycleMethodDailyInput[key];
    }
  });
});
var _PlatformSettlementCycleMethodInput = require("./PlatformSettlementCycleMethodInput.cjs");
Object.keys(_PlatformSettlementCycleMethodInput).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementCycleMethodInput[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementCycleMethodInput[key];
    }
  });
});
var _PlatformSettlementCycleMethodManualDates = require("./PlatformSettlementCycleMethodManualDates.cjs");
Object.keys(_PlatformSettlementCycleMethodManualDates).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementCycleMethodManualDates[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementCycleMethodManualDates[key];
    }
  });
});
var _PlatformSettlementCycleMethodManualDatesInput = require("./PlatformSettlementCycleMethodManualDatesInput.cjs");
Object.keys(_PlatformSettlementCycleMethodManualDatesInput).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementCycleMethodManualDatesInput[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementCycleMethodManualDatesInput[key];
    }
  });
});
var _PlatformSettlementCycleMethodMonthly = require("./PlatformSettlementCycleMethodMonthly.cjs");
Object.keys(_PlatformSettlementCycleMethodMonthly).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementCycleMethodMonthly[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementCycleMethodMonthly[key];
    }
  });
});
var _PlatformSettlementCycleMethodMonthlyInput = require("./PlatformSettlementCycleMethodMonthlyInput.cjs");
Object.keys(_PlatformSettlementCycleMethodMonthlyInput).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementCycleMethodMonthlyInput[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementCycleMethodMonthlyInput[key];
    }
  });
});
var _PlatformSettlementCycleMethodWeekly = require("./PlatformSettlementCycleMethodWeekly.cjs");
Object.keys(_PlatformSettlementCycleMethodWeekly).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementCycleMethodWeekly[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementCycleMethodWeekly[key];
    }
  });
});
var _PlatformSettlementCycleMethodWeeklyInput = require("./PlatformSettlementCycleMethodWeeklyInput.cjs");
Object.keys(_PlatformSettlementCycleMethodWeeklyInput).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementCycleMethodWeeklyInput[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementCycleMethodWeeklyInput[key];
    }
  });
});
var _PlatformSettlementFormula = require("./PlatformSettlementFormula.cjs");
Object.keys(_PlatformSettlementFormula).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementFormula[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementFormula[key];
    }
  });
});
var _PlatformSettlementFormulaError = require("./PlatformSettlementFormulaError.cjs");
Object.keys(_PlatformSettlementFormulaError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementFormulaError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementFormulaError[key];
    }
  });
});
var _PlatformSettlementFormulaInvalidFunction = require("./PlatformSettlementFormulaInvalidFunction.cjs");
Object.keys(_PlatformSettlementFormulaInvalidFunction).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementFormulaInvalidFunction[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementFormulaInvalidFunction[key];
    }
  });
});
var _PlatformSettlementFormulaInvalidOperator = require("./PlatformSettlementFormulaInvalidOperator.cjs");
Object.keys(_PlatformSettlementFormulaInvalidOperator).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementFormulaInvalidOperator[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementFormulaInvalidOperator[key];
    }
  });
});
var _PlatformSettlementFormulaInvalidSyntax = require("./PlatformSettlementFormulaInvalidSyntax.cjs");
Object.keys(_PlatformSettlementFormulaInvalidSyntax).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementFormulaInvalidSyntax[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementFormulaInvalidSyntax[key];
    }
  });
});
var _PlatformSettlementFormulaInvalidVariable = require("./PlatformSettlementFormulaInvalidVariable.cjs");
Object.keys(_PlatformSettlementFormulaInvalidVariable).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementFormulaInvalidVariable[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementFormulaInvalidVariable[key];
    }
  });
});
var _PlatformSettlementFormulaPosition = require("./PlatformSettlementFormulaPosition.cjs");
Object.keys(_PlatformSettlementFormulaPosition).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementFormulaPosition[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementFormulaPosition[key];
    }
  });
});
var _PlatformSettlementFormulaUnexpectedFunctionArguments = require("./PlatformSettlementFormulaUnexpectedFunctionArguments.cjs");
Object.keys(_PlatformSettlementFormulaUnexpectedFunctionArguments).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementFormulaUnexpectedFunctionArguments[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementFormulaUnexpectedFunctionArguments[key];
    }
  });
});
var _PlatformSettlementFormulaUnknownError = require("./PlatformSettlementFormulaUnknownError.cjs");
Object.keys(_PlatformSettlementFormulaUnknownError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementFormulaUnknownError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementFormulaUnknownError[key];
    }
  });
});
var _PlatformSettlementFormulaUnsupportedVariable = require("./PlatformSettlementFormulaUnsupportedVariable.cjs");
Object.keys(_PlatformSettlementFormulaUnsupportedVariable).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementFormulaUnsupportedVariable[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementFormulaUnsupportedVariable[key];
    }
  });
});
var _PlatformSettlementRule = require("./PlatformSettlementRule.cjs");
Object.keys(_PlatformSettlementRule).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformSettlementRule[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformSettlementRule[key];
    }
  });
});
var _PlatformUserDefinedPropertyNotFoundError = require("./PlatformUserDefinedPropertyNotFoundError.cjs");
Object.keys(_PlatformUserDefinedPropertyNotFoundError).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformUserDefinedPropertyNotFoundError[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformUserDefinedPropertyNotFoundError[key];
    }
  });
});
var _PlatformUserDefinedPropertyValue = require("./PlatformUserDefinedPropertyValue.cjs");
Object.keys(_PlatformUserDefinedPropertyValue).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _PlatformUserDefinedPropertyValue[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _PlatformUserDefinedPropertyValue[key];
    }
  });
});
var _ReschedulePlatformAdditionalFeePolicyBody = require("./ReschedulePlatformAdditionalFeePolicyBody.cjs");
Object.keys(_ReschedulePlatformAdditionalFeePolicyBody).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _ReschedulePlatformAdditionalFeePolicyBody[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _ReschedulePlatformAdditionalFeePolicyBody[key];
    }
  });
});
var _ReschedulePlatformAdditionalFeePolicyResponse = require("./ReschedulePlatformAdditionalFeePolicyResponse.cjs");
Object.keys(_ReschedulePlatformAdditionalFeePolicyResponse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _ReschedulePlatformAdditionalFeePolicyResponse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _ReschedulePlatformAdditionalFeePolicyResponse[key];
    }
  });
});
var _ReschedulePlatformContractBody = require("./ReschedulePlatformContractBody.cjs");
Object.keys(_ReschedulePlatformContractBody).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _ReschedulePlatformContractBody[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _ReschedulePlatformContractBody[key];
    }
  });
});
var _ReschedulePlatformContractResponse = require("./ReschedulePlatformContractResponse.cjs");
Object.keys(_ReschedulePlatformContractResponse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _ReschedulePlatformContractResponse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _ReschedulePlatformContractResponse[key];
    }
  });
});
var _ReschedulePlatformDiscountSharePolicyBody = require("./ReschedulePlatformDiscountSharePolicyBody.cjs");
Object.keys(_ReschedulePlatformDiscountSharePolicyBody).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _ReschedulePlatformDiscountSharePolicyBody[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _ReschedulePlatformDiscountSharePolicyBody[key];
    }
  });
});
var _ReschedulePlatformDiscountSharePolicyResponse = require("./ReschedulePlatformDiscountSharePolicyResponse.cjs");
Object.keys(_ReschedulePlatformDiscountSharePolicyResponse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _ReschedulePlatformDiscountSharePolicyResponse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _ReschedulePlatformDiscountSharePolicyResponse[key];
    }
  });
});
var _ReschedulePlatformPartnerBody = require("./ReschedulePlatformPartnerBody.cjs");
Object.keys(_ReschedulePlatformPartnerBody).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _ReschedulePlatformPartnerBody[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _ReschedulePlatformPartnerBody[key];
    }
  });
});
var _ReschedulePlatformPartnerResponse = require("./ReschedulePlatformPartnerResponse.cjs");
Object.keys(_ReschedulePlatformPartnerResponse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _ReschedulePlatformPartnerResponse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _ReschedulePlatformPartnerResponse[key];
    }
  });
});
var _SchedulePlatformAdditionalFeePolicyBody = require("./SchedulePlatformAdditionalFeePolicyBody.cjs");
Object.keys(_SchedulePlatformAdditionalFeePolicyBody).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformAdditionalFeePolicyBody[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformAdditionalFeePolicyBody[key];
    }
  });
});
var _SchedulePlatformAdditionalFeePolicyResponse = require("./SchedulePlatformAdditionalFeePolicyResponse.cjs");
Object.keys(_SchedulePlatformAdditionalFeePolicyResponse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformAdditionalFeePolicyResponse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformAdditionalFeePolicyResponse[key];
    }
  });
});
var _SchedulePlatformContractBody = require("./SchedulePlatformContractBody.cjs");
Object.keys(_SchedulePlatformContractBody).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformContractBody[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformContractBody[key];
    }
  });
});
var _SchedulePlatformContractResponse = require("./SchedulePlatformContractResponse.cjs");
Object.keys(_SchedulePlatformContractResponse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformContractResponse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformContractResponse[key];
    }
  });
});
var _SchedulePlatformDiscountSharePolicyBody = require("./SchedulePlatformDiscountSharePolicyBody.cjs");
Object.keys(_SchedulePlatformDiscountSharePolicyBody).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformDiscountSharePolicyBody[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformDiscountSharePolicyBody[key];
    }
  });
});
var _SchedulePlatformDiscountSharePolicyResponse = require("./SchedulePlatformDiscountSharePolicyResponse.cjs");
Object.keys(_SchedulePlatformDiscountSharePolicyResponse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformDiscountSharePolicyResponse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformDiscountSharePolicyResponse[key];
    }
  });
});
var _SchedulePlatformPartnerBody = require("./SchedulePlatformPartnerBody.cjs");
Object.keys(_SchedulePlatformPartnerBody).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformPartnerBody[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformPartnerBody[key];
    }
  });
});
var _SchedulePlatformPartnerResponse = require("./SchedulePlatformPartnerResponse.cjs");
Object.keys(_SchedulePlatformPartnerResponse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformPartnerResponse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformPartnerResponse[key];
    }
  });
});
var _SchedulePlatformPartnersBody = require("./SchedulePlatformPartnersBody.cjs");
Object.keys(_SchedulePlatformPartnersBody).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformPartnersBody[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformPartnersBody[key];
    }
  });
});
var _SchedulePlatformPartnersBodyUpdate = require("./SchedulePlatformPartnersBodyUpdate.cjs");
Object.keys(_SchedulePlatformPartnersBodyUpdate).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformPartnersBodyUpdate[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformPartnersBodyUpdate[key];
    }
  });
});
var _SchedulePlatformPartnersBodyUpdateAccount = require("./SchedulePlatformPartnersBodyUpdateAccount.cjs");
Object.keys(_SchedulePlatformPartnersBodyUpdateAccount).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformPartnersBodyUpdateAccount[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformPartnersBodyUpdateAccount[key];
    }
  });
});
var _SchedulePlatformPartnersBodyUpdateContact = require("./SchedulePlatformPartnersBodyUpdateContact.cjs");
Object.keys(_SchedulePlatformPartnersBodyUpdateContact).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformPartnersBodyUpdateContact[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformPartnersBodyUpdateContact[key];
    }
  });
});
var _SchedulePlatformPartnersBodyUpdateType = require("./SchedulePlatformPartnersBodyUpdateType.cjs");
Object.keys(_SchedulePlatformPartnersBodyUpdateType).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformPartnersBodyUpdateType[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformPartnersBodyUpdateType[key];
    }
  });
});
var _SchedulePlatformPartnersBodyUpdateTypeBusiness = require("./SchedulePlatformPartnersBodyUpdateTypeBusiness.cjs");
Object.keys(_SchedulePlatformPartnersBodyUpdateTypeBusiness).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformPartnersBodyUpdateTypeBusiness[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformPartnersBodyUpdateTypeBusiness[key];
    }
  });
});
var _SchedulePlatformPartnersBodyUpdateTypeNonWhtPayer = require("./SchedulePlatformPartnersBodyUpdateTypeNonWhtPayer.cjs");
Object.keys(_SchedulePlatformPartnersBodyUpdateTypeNonWhtPayer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformPartnersBodyUpdateTypeNonWhtPayer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformPartnersBodyUpdateTypeNonWhtPayer[key];
    }
  });
});
var _SchedulePlatformPartnersBodyUpdateTypeWhtPayer = require("./SchedulePlatformPartnersBodyUpdateTypeWhtPayer.cjs");
Object.keys(_SchedulePlatformPartnersBodyUpdateTypeWhtPayer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformPartnersBodyUpdateTypeWhtPayer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformPartnersBodyUpdateTypeWhtPayer[key];
    }
  });
});
var _SchedulePlatformPartnersResponse = require("./SchedulePlatformPartnersResponse.cjs");
Object.keys(_SchedulePlatformPartnersResponse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _SchedulePlatformPartnersResponse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _SchedulePlatformPartnersResponse[key];
    }
  });
});
var _UpdatePlatformAdditionalFeePolicyBody = require("./UpdatePlatformAdditionalFeePolicyBody.cjs");
Object.keys(_UpdatePlatformAdditionalFeePolicyBody).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _UpdatePlatformAdditionalFeePolicyBody[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _UpdatePlatformAdditionalFeePolicyBody[key];
    }
  });
});
var _UpdatePlatformBody = require("./UpdatePlatformBody.cjs");
Object.keys(_UpdatePlatformBody).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _UpdatePlatformBody[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _UpdatePlatformBody[key];
    }
  });
});
var _UpdatePlatformBodySettlementFormula = require("./UpdatePlatformBodySettlementFormula.cjs");
Object.keys(_UpdatePlatformBodySettlementFormula).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _UpdatePlatformBodySettlementFormula[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _UpdatePlatformBodySettlementFormula[key];
    }
  });
});
var _UpdatePlatformBodySettlementRule = require("./UpdatePlatformBodySettlementRule.cjs");
Object.keys(_UpdatePlatformBodySettlementRule).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _UpdatePlatformBodySettlementRule[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _UpdatePlatformBodySettlementRule[key];
    }
  });
});
var _UpdatePlatformContractBody = require("./UpdatePlatformContractBody.cjs");
Object.keys(_UpdatePlatformContractBody).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _UpdatePlatformContractBody[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _UpdatePlatformContractBody[key];
    }
  });
});
var _UpdatePlatformDiscountSharePolicyBody = require("./UpdatePlatformDiscountSharePolicyBody.cjs");
Object.keys(_UpdatePlatformDiscountSharePolicyBody).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _UpdatePlatformDiscountSharePolicyBody[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _UpdatePlatformDiscountSharePolicyBody[key];
    }
  });
});
var _UpdatePlatformPartnerBody = require("./UpdatePlatformPartnerBody.cjs");
Object.keys(_UpdatePlatformPartnerBody).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _UpdatePlatformPartnerBody[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _UpdatePlatformPartnerBody[key];
    }
  });
});
var _UpdatePlatformPartnerBodyAccount = require("./UpdatePlatformPartnerBodyAccount.cjs");
Object.keys(_UpdatePlatformPartnerBodyAccount).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _UpdatePlatformPartnerBodyAccount[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _UpdatePlatformPartnerBodyAccount[key];
    }
  });
});
var _UpdatePlatformPartnerBodyContact = require("./UpdatePlatformPartnerBodyContact.cjs");
Object.keys(_UpdatePlatformPartnerBodyContact).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _UpdatePlatformPartnerBodyContact[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _UpdatePlatformPartnerBodyContact[key];
    }
  });
});
var _UpdatePlatformPartnerBodyType = require("./UpdatePlatformPartnerBodyType.cjs");
Object.keys(_UpdatePlatformPartnerBodyType).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _UpdatePlatformPartnerBodyType[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _UpdatePlatformPartnerBodyType[key];
    }
  });
});
var _UpdatePlatformPartnerBodyTypeBusiness = require("./UpdatePlatformPartnerBodyTypeBusiness.cjs");
Object.keys(_UpdatePlatformPartnerBodyTypeBusiness).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _UpdatePlatformPartnerBodyTypeBusiness[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _UpdatePlatformPartnerBodyTypeBusiness[key];
    }
  });
});
var _UpdatePlatformPartnerBodyTypeNonWhtPayer = require("./UpdatePlatformPartnerBodyTypeNonWhtPayer.cjs");
Object.keys(_UpdatePlatformPartnerBodyTypeNonWhtPayer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _UpdatePlatformPartnerBodyTypeNonWhtPayer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _UpdatePlatformPartnerBodyTypeNonWhtPayer[key];
    }
  });
});
var _UpdatePlatformPartnerBodyTypeWhtPayer = require("./UpdatePlatformPartnerBodyTypeWhtPayer.cjs");
Object.keys(_UpdatePlatformPartnerBodyTypeWhtPayer).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _UpdatePlatformPartnerBodyTypeWhtPayer[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _UpdatePlatformPartnerBodyTypeWhtPayer[key];
    }
  });
});
var _UpdatePlatformResponse = require("./UpdatePlatformResponse.cjs");
Object.keys(_UpdatePlatformResponse).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _UpdatePlatformResponse[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _UpdatePlatformResponse[key];
    }
  });
});
var _client = require("./client.cjs");
Object.keys(_client).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _client[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _client[key];
    }
  });
});
var _policy = _interopRequireWildcard(require("./policy/index.cjs"));
exports.policy = _policy;
var _partner = _interopRequireWildcard(require("./partner/index.cjs"));
exports.partner = _partner;
var _transfer = _interopRequireWildcard(require("./transfer/index.cjs"));
exports.transfer = _transfer;
var _partnerSettlement = _interopRequireWildcard(require("./partnerSettlement/index.cjs"));
exports.partnerSettlement = _partnerSettlement;
var _payout = _interopRequireWildcard(require("./payout/index.cjs"));
exports.payout = _payout;
var _bulkPayout = _interopRequireWildcard(require("./bulkPayout/index.cjs"));
exports.bulkPayout = _bulkPayout;
var _account = _interopRequireWildcard(require("./account/index.cjs"));
exports.account = _account;
var _accountTransfer = _interopRequireWildcard(require("./accountTransfer/index.cjs"));
exports.accountTransfer = _accountTransfer;
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
