// @portone/server-sdk@0.8.1/dist/generated/platform/partnerSettlement/client.mjs
import { PartnerSettlementError } from "./PartnerSettlementError.mjs";
import { USER_AGENT } from "../../../client.mjs";
export function PartnerSettlementClient(init) {
  const baseUrl = init.baseUrl ?? "https://api.portone.io";
  const secret = init.secret;
  return {
    getPlatformPartnerSettlements: async (options) => {
      const {
        page,
        filter,
        isForTest
      } = options;
      const requestBody = JSON.stringify({
        page,
        filter,
        isForTest
      });
      const query = [
        ["requestBody", requestBody]
      ].flatMap(([key, value]) => value == null ? [] : `${key}=${encodeURIComponent(value)}`).join("&");
      const response = await fetch(
        new URL(`/platform/partner-settlements?${query}`, baseUrl),
        {
          method: "GET",
          headers: {
            Authorization: `PortOne ${secret}`,
            "User-Agent": USER_AGENT
          }
        }
      );
      if (!response.ok) {
        throw new GetPlatformPartnerSettlementsError(await response.json());
      }
      return response.json();
    }
  };
}
export class GetPlatformPartnerSettlementsError extends PartnerSettlementError {
  /** @ignore */
  constructor(data) {
    super(data);
    Object.setPrototypeOf(this, GetPlatformPartnerSettlementsError.prototype);
    this.name = "GetPlatformPartnerSettlementsError";
  }
}
